proxy/http2/Http2DependencyTree.h

```diff
@@ -130,8 +150,32 @@ template <typename T> class Tree
   Node *_root = new Node(this);
   uint32_t _max_depth;
   uint32_t _node_count = 0;
+  /*
+   * _ancestors in a circular buffer tracking parent relationships for
+   * recently completed nodes.  Without this new streams may not find their
+   * parents and be inserted at the root, violating the client's desired
+   * dependency relationship.
```

Comment:
Just a comment: RFC 7540 says below in [5.3.4.  Prioritization State Management](https://tools.ietf.org/html/rfc7540#section-5.3.4). So circular buffer make sense.

> To avoid these problems, an endpoint SHOULD retain stream
   prioritization state for a period after streams become closed.